tools/plugins/com.liferay.ide.project.core/src/com/liferay/ide/project/core/modules/ComponentNameValidationService.java

```diff
@@ -0,0 +1,64 @@
+/*******************************************************************************
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ *
+ *******************************************************************************/
+package com.liferay.ide.project.core.modules;
+
+import com.liferay.ide.core.util.CoreUtil;
+
+import org.eclipse.core.runtime.IStatus;
+import org.eclipse.jdt.core.JavaConventions;
+import org.eclipse.jdt.internal.compiler.impl.CompilerOptions;
+import org.eclipse.sapphire.modeling.Status;
+import org.eclipse.sapphire.services.ValidationService;
+
+
+
+/** 
+ * @author Simon Jiang
+ */
+@SuppressWarnings( "restriction" )
+public class ComponentNameValidationService extends ValidationService
```

Comment:
Lets add a ComponentNameDefaultValueService that makes the component name be something like this:
<Projectname>Portlet.java